### PR TITLE
Update machineconfig to enable required annotations for flexible instance types

### PR DESCRIPTION
**Pull request by bpradipt**
This PR updates the machineconfig to deploy updated /opt/kata/configuration-remote.toml to enable required annotations to support
flexible instance types for peerpods

Fixes: #[KATA-2268](https://issues.redhat.com//browse/KATA-2268)

### CODE REVIEW

File: config/peerpods/mc-40-kata-remote-config.yaml [hunk 1]
```diff
@@ -3,9 +3,9 @@ apiVersion: machineconfiguration.openshift.io/v1
 kind: MachineConfig
 metadata:
   labels:
-    app: peerpodconfig-openshift
+    app: example-kataconfig
```
Comment (pmores) [inline config/peerpods/mc-40-kata-remote-config.yaml:6]: Is this intended?

File: config/peerpods/mc-40-kata-remote-config.yaml [hunk 2]
```diff
@@ -3,9 +3,9 @@ apiVersion: machineconfiguration.openshift.io/v1
...
     machineconfiguration.openshift.io/role: kata-oc
-  name: 40-worker-kata-remote-config
+  name: mc-40-kata-remote-config.yaml
```
Comment (pmores) [inline config/peerpods/mc-40-kata-remote-config.yaml:8]: This diverges from the usual MC naming convention, can you explain the purpose of this move?  I suspect that given that MC names typically start with numbers, this would ensure the kata-remote-config MC is always applied last.

File: config/peerpods/mc-40-kata-remote-config.yaml [hunk 3]
```diff
@@ -14,6 +14,6 @@ spec:
...
...2ij6gzwYfXcjGqYLlOx4zcQvHbr3GgSSAqM0mkQPDfS5EgIccaSCGJWySwsQLvt4q3IKE4IqinCclz6lTMyWWnb4ncYD6NE1jOtLn172C8F4+OWg8JHC+NemeBtg9CWHB6CHCM9bGfuDOfNMRmcYwe9bOIA2KkJtTGFyqkB9pkEk0I3uBvefp8mxudm3MAkIrhjBFbJdqD/x61urOCCWSVbNphnewEM7GgPHWdzvaq10saISohMqvpxauHgD0FcuKFoaA4U4seYUvYvg1TIYcFw44Guiw2tMJB7c+aTKYcUWx4eG9c6rWv2Dgw7ovRoEdeJOB6/S6CJi0SMQLrkPqrtBhIO0sQ8Sl5iFzpWw495PfSCslV58OppY43WXwy8hPGTQn1KuZBjfw11rBr7QgEmNvcUg3PFlW0SZ33eJxOAb1n786ioaG7p2dOQ0R7FyQReJrKJelzo9Zlor52chE3shPmQym4lsnX2kdtwjE5iW6R/kOHC7D2gKm7lvzhQdRYmharvnrl0L8hbD5lZ6tVa18vt+r62FVtr0WMRabWISQKy7hjTX2/Qe0LGtUC9Wr2A/fDydDIcFYhzND6CibtFhFdSOxfqCTPr2cYqcY16A2CBE1otjOn3ap48QZC9+6SmpJ0FxxxtKgd/6dJgLMTZ5Ccw4oIc3RMUefJKxOHonXQi56ZnYtpY5VHyHRDx/JV6+uhHv3r4QVzf/ZfKBilUfR1kovrYQOIpjl1dlySk4DLLeXIUhjdRyQ4jE0ZGbf+yd9mOUCaNXJoAdpiTHWBSQXvG9hb1jIUBmCK7NE3L5g25RpTiEHNqPceiPOL/vyGAwQNdUqdXjhF59jacrWUPSo2CcR6OkkOThpL1O0YTxS8o4vXMdt5jGHEbssnf9IOk9uOjhJw1HzB2fR+bWdINH7EG8e/MTFwzq6PT2pEXiqALojiB1Q4KvTBE7L5Tmfjr8N71tEC6/7UO+fEiOYWthv78Zf2hAOn6QACU0bICrv8XFZJp2cFlILy+WIVhY3sJ+iVAGi0UFG59a+BytQ1omRb7oB9GS4oTIesoMEt7qlf1bmniYI7Tooe7lMl2aWSu9ZAzsUulliA8PKJJObMApi/7daKY/Q3FcdSNEERniEgy43uLpjVCJrkCzn9/Cfr625hbsojRm6UFL7Zd/4b/nqvh+8n8DAAD//8T7BEz6XAAA
```
Comment (pmores) [inline config/peerpods/mc-40-kata-remote-config.yaml:17]: Could you add some explanation why this is changing?

File: config/peerpods/kata-remote.conf [hunk 4]
```diff
@@ -13,7 +13,8 @@
 
 [hypervisor.remote]
 remote_hypervisor_socket = "/run/peerpod/hypervisor.sock"
-remote_hypervisor_timeout = 600
+remote_hypervisor_timeout = 900
+disable_guest_selinux = true
```
Comment (pmores) [inline config/peerpods/kata-remote.conf:17]: Are these changes related to enabling annotations processing?

File: config/peerpods/mc-40-kata-remote-config.yaml (hunk 2 above)
Comment (bpradipt) [reply config/peerpods/mc-40-kata-remote-config.yaml:8]: My bad. The conversion using butane tool uses a config and I used an old config it seems. I'll commit the butane config as well

File: config/peerpods/kata-remote.conf (hunk 4 above)
Comment (bpradipt) [reply config/peerpods/kata-remote.conf:17]: The kata-remote.conf that's committed here is not matching with the one deployed. Hence I updated it to the latest. I'll also commit the configuration used to create the machineconfig from the original file to avoid any future inconsistency

Comment (pmores) [reply config/peerpods/kata-remote.conf:17]: Fair enough, makes sense.  I wasn't sure if the changes are related since they are parts of the same commit.

Comment (bpradipt) [reply config/peerpods/kata-remote.conf:17]: I'll double check as well just to be sure.

File: config/peerpods/kata-remote.conf [hunk 5]
```diff
@@ -536,6 +537,7 @@ disable_new_netns = true
 # The sandbox cgroup is constrained if there is no container type annotation.
 # See: https://godoc.org/github.com/kata-containers/runtime/virtcontainers#ContainerType
 sandbox_cgroup_only=false
+static_sandbox_resource_mgmt = true
```
[the remark this answers is not in the document]
Comment (bpradipt) [reply config/peerpods/kata-remote.conf:540]: Peerpods don't support hot plugging. Hence the new feature is "flexible" :-)

File: config/peerpods/mc-40-kata-remote-config.yaml (hunk 3 above)
Comment (bpradipt) [reply config/peerpods/mc-40-kata-remote-config.yaml:17]: yes

Comment (bpradipt) [reply config/peerpods/mc-40-kata-remote-config.yaml:17]: I added the original config toml file and added details about  how the machineconfig is created. Hope it helps. Let me know if you want me to add any additional info.

File: config/peerpods/kata-remote.conf (hunk 4 above)
Comment (bpradipt) [reply config/peerpods/kata-remote.conf:17]: Using the following cmd on the original file will get the plaintext configuration toml
```
grep source: mc-40-kata-remote-config.yaml | cut -d "," -f2 | base64 -d | gunzip
```

File: config/peerpods/kata-remote.conf (hunk 5 above)
[the remark this answers is not in the document]
Comment (bpradipt) [reply config/peerpods/kata-remote.conf:540]: Previously peer-pods only supported a single cloud instance type and new feature allows user the flexibility to specify the type of cloud instance they need for their pod.

**Comment by bpradipt** [comment]
@jensfr @pmores @snir911 ptal. Once this is merged, it'll be possible for QE to test this functionality via OSC

**Review by pmores — APPROVED** [review]
lgtm, thanks @bpradipt !

File: config/peerpods/local/configuration-remote.toml [hunk 6]
```diff
@@ -13,7 +13,8 @@
 
 [hypervisor.remote]
 remote_hypervisor_socket = "/run/peerpod/hypervisor.sock"
-remote_hypervisor_timeout = 600
+remote_hypervisor_timeout = 900
+disable_guest_selinux = true
```
Comment (snir911) [inline config/peerpods/local/configuration-remote.toml:17]: why is this needed?

**Review by snir911 — APPROVED** [review]
LGTM overlall, added minor comment

File: config/peerpods/local/configuration-remote.toml (hunk 6 above)
Comment (bpradipt) [reply config/peerpods/local/configuration-remote.toml:17]: This is there in the current config. In tests we have seen sometimes the timeout required is higher. Also selinux is disabled in the guest for now to avoid issues with accessing the TPM device. More work is needed to enable selinux for the Pod VM
